src/Patch/Document.js

```diff
@@ -71,6 +71,16 @@ export default function(internals) {
       return result;
     });
 
+  Object.defineProperty(document, 'body', {
+    get: function() {
+      return document.querySelector('body');
+    },
+    set: function(body) {
+      document.documentElement.replaceChild(body, document.body);
```

Comment:
The reason this patch works is because [`Node.prototype.replaceChild` is patched](https://github.com/webcomponents/custom-elements/blob/6ad9993949a4f25c05b50b27a031a2ebfc49df89/src/Patch/Node.js#L122), which is being called here. I think it would be better if these patches weren't reentrant, `replaceChild` seems like a reasonable point to look for starters on how to implement this. I think this one will be a bit simpler because you don't have to worry about DocumentFragments (which will throw) or multiple nodes being inserted / removed at once (since it's a setter).

Reply:
I thought this would be more elegant than duplicating code. But sure, I can also go the non-reentrant route here.